source/loader/layers/sanitizer/asan/asan_ddi.cpp

```diff
@@ -1443,8 +1422,8 @@ __urdlllocal ur_result_t UR_APICALL urKernelSetArgValue(
     std::shared_ptr<KernelInfo> KernelInfo;
     if (argSize == sizeof(ur_mem_handle_t) &&
         (MemBuffer = getAsanInterceptor()->getMemBuffer(
-             *ur_cast<const ur_mem_handle_t *>(pArgValue))) &&
-        (KernelInfo = getAsanInterceptor()->getKernelInfo(hKernel))) {
+             *ur_cast<const ur_mem_handle_t *>(pArgValue)))) {
+        auto KernelInfo = getAsanInterceptor()->getKernelInfo(hKernel);
```

Comment:
this is aliasing an already existing variable that appears to be otherwise unused:
`std::shared_ptr<KernelInfo> KernelInfo;` << I think you should be able to just remove this.